<launch>
  <arg name="initial_pose_x" default="-3.0"/>
  <arg name="initial_pose_y" default="1.0"/>
  <arg name="initial_pose_a" default="0.0"/>

  <!-- Set tf -->
  <node pkg="tf" type="static_transform_publisher" name="world_to_map" args="0 0 0 0 0 0 world map 100" />

  <!-- parameters -->
  <rosparam>
    vi_node:
      action_list:
        - name: forward
          onestep_forward_m: 0.3
          onestep_rotation_deg: 0.0
        - name: back
          onestep_forward_m: -0.2
          onestep_rotation_deg: 0.0
        - name: right
          onestep_forward_m: 0.0
          onestep_rotation_deg: -20.0
        - name: rightfw
          onestep_forward_m: 0.2
          onestep_rotation_deg: -20.0
        - name: left
          onestep_forward_m: 0.0
          onestep_rotation_deg: 20.0
        - name: leftfw
          onestep_forward_m: 0.2
          onestep_rotation_deg: 20.0
  </rosparam>

  <!-- Arguments -->
  <arg name="model" default="burger" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find value_iteration)/maps/house.yaml"/>
  <arg name="open_rviz" default="true"/>

  <!-- Turtlebot3 -->
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include>

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>
<!-- 
  <node pkg="map_server" name="cost_map" type="map_server" args="$(find value_iteration)/maps/cost.yaml">
    <remap from="static_map" to="cost_map" />
    <remap from="map" to="cost_map_for_vi" />
  </node>
-->

  <!-- MCL -->
  <include file="$(find value_iteration)/launch/emcl.launch"/>
  <!-- <include file="$(find value_iteration)/launch/amcl.launch"/> -->

  <!-- Launch other launch files -->
  <include file="$(find turtlebot3_gazebo)/launch/turtlebot3_house.launch" />

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find value_iteration)/config/conf.rviz"/>
  </group>

  <!-- Launch the value_iteration node -->
  <node pkg="value_iteration" name="vi_node" type="vi_node" output="screen" required="true">
    <param name="online" value="true" />
    <param name="theta_cell_num" value="60" />
    <param name="thread_num" value="1" />
    <param name="safety_radius" value="0.2" />
    <param name="safety_radius_penalty" value="100000" />
    <param name="goal_margin_radius" value="0.3" />
    <param name="goal_margin_theta" value="15" />
    <param name="map_type" value="occupancy" />
<!--    <param name="map_type" value="cost" /> -->

  </node>
  <node pkg="value_iteration" name="vi_controller_turtle_env" type="vi_controller.py" output="screen" required="true" />

</launch>
